### PR TITLE
Fix/popup

### PR DESCRIPTION
**Pull request by AlexRutledge1**
# Describe your changes
The NativeBase Modal component doesn't work properly in React, it seems like it's an issue with the View component. To circumvent this I installed reactjs-popup and replaced the aforementioned Modal's with the Popup component.
How the new Popup component looks: 
image
We are still able to use NativeBase components within the Popup, so this won't affect the consistency of our style.
# Issue ticket number and link
#42

### CODE REVIEW

File: src/components/Profile/ProfileBanner.tsx [hunk 1]
```diff
@@ -37,27 +40,22 @@ const ProfileBanner = ({user}: {user: User | undefined}) => {
...
-            </Modal>
+            <Popup open={editPermissionModal} closeOnDocumentClick={false} 
+                onClose={() => setEditPermissionModal(false)} modal>
+                <Box p={4} w='100%' h='100%' borderRadius='sm'>
+                    <Text alignSelf='center' bold fontSize='lg'>Edit user permission level</Text>
+                    <Select defaultValue={'' + user?.status}>
+                        <Select.Item label='Unverified (cannot login)' value='0'/>
+                        <Select.Item label='Verified (can login, but cannot post)' value='1'/>
+                        <Select.Item label='Approved (can post)' value='2'/>
+                        {userAuth === 4 && <Select.Item label='Employee' value='3'/>}
```
Comment (DanielW48) [inline src/components/Profile/ProfileBanner.tsx]: Do we want to add the manager switch in this popup as well?

File: src/pages/Signup.tsx [hunk 2]
```diff
@@ -58,26 +60,24 @@ const Signup =  () => {
...
-                    <Modal.CloseButton/>
-                    <Modal.Header>Email Confirmation</Modal.Header>
-                    <Modal.Body>
+            <Popup modal open={showModal} onClose={closeModal}>
+                <Box flex={1} alignSelf='center' alignItems={'center'} justifyContent='center' paddingTop= '1vh'>
+                    <Text variant = {'header'}>
+                        Email Confirmation
+                    </Text>
+                    <Text variant = {'body'}>
                         A confirmation email has been sent to verify your email address.{'\n'}
```
Comment (DanielW48) [inline src/pages/Signup.tsx:69]: Would this popup look better with centered text?

**Review by DanielW48 — APPROVED** [review]
LGTM

File: src/components/Profile/ProfileBanner.tsx (hunk 1 above)
Comment (AlexRutledge1) [reply src/components/Profile/ProfileBanner.tsx]: Yeah we should, but I think I might want to reformat the selection process to make it more intuitive. I'm gonna work on it for a bit then ping you when I want you to look over it.

**Comment by AlexRutledge1** [comment]
The text in the signup popup won't center for some reason, but I think it looks best aligned to the left anyway. As for the user permission popup, I made some changes to make its functionality clearer:
- If the current logged in user is a manager, and the profile being viewed is an employee, display a button to make the profile the manager account. In the screenshot I manually changed the states so that I could show how it will look for employees.
- Made a radio that will set whether the employee wants to demote or promote the profile
- Depending upon which radio is selected and which status the profile has, a dropdown will display the possible actions to be taken
- Include a text field where employees can describe the reason for the action.
image

**Comment by DanielW48** [comment]
Yeah money, send it

**Pull request merged**